src/bosh_azure_cpi/lib/cloud/azure/models/instance_id_parser.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Bosh::AzureCloud
+  class InstanceIdParser
```

Comment:
`InstanceIdParser` is a little confusing. Suggest changing it to something like CidParser

Reply:
good.